services/Skyra.Cdn/Controllers/CdnController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Skyra.Database;
+using Skyra.Database.Models.Entities;
+
+namespace Skyra.Cdn.Controllers
+{
+	[ApiController]
+	[Route("[controller]")]
+	public class CdnController : ControllerBase
+	{
+		private readonly IDatabase _database;
+
+		public CdnController(IDatabase database)
+		{
+			_database = database;
+		}
+
+		[HttpGet("{id}")]
+		[ResponseCache(Duration = 60 * 60, Location = ResponseCacheLocation.Client, NoStore = false)]
+		public async Task<IActionResult> Get(string id)
+		{
+			var result = await _database.GetAssetAsync(id);
+			if (!result.Success)
+			{
+				return NotFound();
+			}
+
+			var asset = result.Value!;
```

Comment:
Use an actual null check here `if(x is not null)`

Reply:
`Response<T>.Value` is `T?`, being `null` when `Success` is false. Here, `Success` is true, and therefore, it cannot be null.

Reply:
Hmmm... I wonder if we should change the design of this. Alright